packages/design-system/src/molecules/tokenSelectContentV2/index.tsx

```diff
@@ -0,0 +1,193 @@
+import { InputV2 } from '..'
+import { isAddress } from '../../utils'
+
+const MODAL_BORDER_RADIUS = 16
+const MODAL_MAX_WIDTH = 380
+const MODAL_MAX_HEIGHT = 552
+
+const styles = StyleSheet.create({
+  modalContainer: {
+    borderRadius: MODAL_BORDER_RADIUS,
+    maxWidth: MODAL_MAX_WIDTH,
+    maxHeight: MODAL_MAX_HEIGHT,
+    overflow: 'hidden',
+  },
+})
+
+export type TokenSelectOptionV2 = {
+  name: string
+  symbol: string
+  logo?: string
+  value: string
+  address: string
+}
+
+interface TokenSelectOptionProps extends TokenSelectOptionV2 {
+  onChange?: (value: string) => void
+  onClose?: () => void
+}
+
+export const TokenSelectOptionV2: React.FC<TokenSelectOptionProps> = ({
+  value,
+  name,
+  symbol,
+  logo,
+  onChange,
+  onClose,
+}) => {
+  const theme = useThemeV2()
+  const handlePressOption = useCallback(() => {
+    onChange?.(value)
+    onClose?.()
+  }, [onChange, value, onClose])
+  return (
+    <Pressable
+      key={value}
+      style={[
+        container.row,
+        container.justifySpaceBetween,
+        container.fullWidth,
+        container.alignCenter,
+        padding.p8,
+      ]}
+      onPress={handlePressOption}
+      testID={`tokenSelectOption__${value}`}
+    >
+      <View style={[container.row, container.alignCenter]}>
+        <CircleImg size={32} uri={logo} />
+        <TypographyV2
+          variant="caption1"
+          style={margin.ml16}
+          color={theme.text.black[100]}
+        >
+          {symbol}
+        </TypographyV2>
+      </View>
+      <TypographyV2 variant="text3" color={theme.text.black[80]}>
+        {name}
+      </TypographyV2>
+    </Pressable>
+  )
+}
+
+export interface TokenSelectContentV2Props {
+  defaultValue?: string
+  withCloseIcon?: boolean
+  options: TokenSelectOptionV2[]
+  label?: string
+  placeholder?: string
+  searchPlaceholder?: string
+  notFoundText?: string
+  onChange: (value: string) => void
+  onClose?: () => void
+  isLoading?: boolean
+  searchToken?: (value: string) => void
+}
+export const TokenSelectContentV2: React.FC<TokenSelectContentV2Props> = ({
+  options,
+  withCloseIcon = false,
+  label = 'Search name or paste contract address',
+  placeholder = 'Select a token',
+  searchPlaceholder = 'Search name or paste address',
+  notFoundText = 'No results found',
+  isLoading = false,
+  onChange,
+  onClose,
+  searchToken,
+}) => {
+  const theme = useThemeV2()
+  const [searchInputValue, setSearchInputValue] = useState('')
+
+  const filteredOptions = useMemo(
+    () =>
+      options.filter(
+        (option) =>
+          option.symbol
+            .toLowerCase()
+            .includes(searchInputValue.toLowerCase()) ||
+          option.name.toLowerCase().includes(searchInputValue.toLowerCase()) ||
+          option.address.toLowerCase().includes(searchInputValue.toLowerCase()),
+      ),
+    [options, searchInputValue],
+  )
+
+  useEffect(() => {
+    if (isAddress(searchInputValue) && filteredOptions.length === 0) {
```

Comment:
If we extract `isEmpty` to a variable:

```suggestion
    if (isAddress(searchInputValue) && isEmpty) {
```